src/components/notifications/NotificationsLayout/UnsubscribeModal/index.tsx

```diff
@@ -8,29 +8,38 @@ import CrossIcon from '../../../general/Icon/CrossIcon'
 import { Modal } from '../../../general/Modal/Modal'
 import './UnsubscribeModal.scss'
 import Text from '../../../general/Text'
+import Spinner from '../../../general/Spinner'
+import { showErrorMessageToast, showSuccessMessageToast } from '../../../../utils/toasts'
+import { useLocation } from 'react-router-dom'
 
 export const UnsubscribeModal: React.FC = () => {
-  const { mode } = useContext(SettingsContext)
-  const themeColors = useColorModeValue(mode)
+  const location = useLocation()
   const { activeSubscriptions, pushClientProxy } = useContext(W3iContext)
   const { unsubscribeModalAppId } = useModals()
+  const [loading, setLoading] = useState(false)
 
   const app = useMemo(
     () => activeSubscriptions.find(activeApp => activeApp.topic === unsubscribeModalAppId),
     [unsubscribeModalAppId]
   )
 
+  useEffect(() => {
+    if (loading) {
```

Comment:
This feels like a weird way to proxy the unsubscribing behaviour. I feel like this can result in:
* loading is set to `true` below, shows success toast
* then `deleteSubscription` throws, shows error toast

Also how come only `location` is in the dependency array?

cc @Cali93 @devceline for second opinion here pls

Reply:
I agree that this was not a best practice. Made some changes.

Redirecting the route and unsubscribing logic is now in the same file. I think this will make more sense.

```
pushClientProxy.observeOne('notify_delete', {
  next: () => {
    unsubscribeModalService.closeModal()
    showSuccessMessageToast(
      `Successfully unsubscribed from ${app ? app.metadata.name : `dapp`}`
     )
    setLoading(false)
    navigate('/notifications/new-app')
  }
})
```